### ISSUE TITLE
VCH appliance prevents the host from entering maintenance mode

### ISSUE THREAD

**Issue by mhagen-vmware** [vmware/vic#2245]
After installing VCH into a nested ESX host environment, running:

govc host.maintenance.enter

now hangs at 100% and doesn't complete.  When I ctrl-c out of the command, it causes the host to be in some sort of invalid state that I cannot exit or leave maintenance mode nor do anything with VMs.  Power cycling the host recovers and the host ends up in maintenance mode when it comes back up.

VIC support bundle:
[container-logs.zip](https://github.com/vmware/vic/files/457052/container-logs.zip)

ESX bundle:

[esx-bundle.zip](https://github.com/vmware/vic/files/457067/esx-bundle.zip)

**Comment by cgtexmex** [comment]
In the case of a standalone host you must first power down the VCH (or any other VMs) before entering maintenance mode.  The govc library could be improved to handle this more gracefully, but it's not an issue w/vic or the VCH.

In a clustered vSphere environment with DRS set to automatic I was able to put a host into maintenance mode which hosted a running VCH.  DRS will first migrate the VCH to another host in the cluster and then enter maintenance mode.  Additionally, I tested with a host that was hosting a running container VM -- same result a successful migration and entry to maintenance mode.

If we want to address the govc improvement then we should open an issue against that project.
